apm-lambda-extension/extension/apm_server.go

```diff
@@ -28,28 +28,37 @@ import (
 
 // todo: can this be a streaming or streaming style call that keeps the
 //       connection open across invocations?
-func PostToApmServer(postBody []byte, config *extensionConfig) error {
+func PostToApmServer(agentData AgentData, config *extensionConfig) error {
 	endpointUri := "intake/v2/events"
-	var compressedBytes bytes.Buffer
-	w := gzip.NewWriter(&compressedBytes)
-	w.Write(postBody)
-	w.Write([]byte{10})
-	w.Close()
+	var req *http.Request
+	var err error
 
-	client := &http.Client{}
-
-	req, err := http.NewRequest("POST", config.apmServerUrl+endpointUri, bytes.NewReader(compressedBytes.Bytes()))
-	if err != nil {
-		return fmt.Errorf("failed to create a new request when posting to APM server: %v", err)
+	if agentData.ContentEncoding == "" {
+		var compressedBytes bytes.Buffer
+		w := gzip.NewWriter(&compressedBytes)
```

Comment:
@felixbarny Should we use gzip to compress the agent data that is not already compressed when sending to the apm server?

Reply:
GZip is fine but we should compress with the best speed compression level

Reply:
done